src/visGeometry/index.ts

```diff
@@ -120,13 +120,13 @@ class VisGeometry {
     public agentPaths: Map<number, AgentPath>;
     public mlogger: ILogger;
     // this is the threejs object that issues all the webgl calls
-    public threejsrenderer: WebGLRenderer;
+    public threejsrenderer!: WebGLRenderer;
```

Comment:
Just because I don't know: is using this operator to delay assignment considered best practice if you're confident that its going to be assigned, or is it hacky as it circumvents a kind of check that typescript does? It's fine with me, just wondering.

Reply:
it's kinda both.  In this case I could have done a typescript definition like `WebGLRenderer | undefined` but the point here is that any usage of that variable before component mount time (before reparent is called) is really an error and totally unexpected, and should have an explicit check.

Reply:
another possible way to handle it is to do `WebGLRenderer | null` and explicitly init it to null in the constructor.  That way it is at least clear that we intentionally did this.  Arguably that is a more robust / readable way to do it.